src/mscorlib/shared/System/Span.cs

```diff
@@ -409,6 +409,9 @@ public T[] ToArray()
         /// </summary>
         public static Span<T> Empty => default(Span<T>);
 
+        // This exposes the internal representation for Span-related apis use only.
+        internal ByReference<T> Pointer => _pointer;
```

Comment:
This should return `ref T`. `ByReference<T>` should be used only for the fields.

Reply:
Why is that?

Wouldn't this make the implementation, essentially a duplicate, but internal, version of DangerousGetPinnableReference?

i.e.
```C#
internal ref T Reference => ref _pointer.Value;
```

Do we then just return `Reference` from MemoryMarshal.GetReference()?